ui/src/infra-model/velho/velho-file-list.tsx

```diff
@@ -3,40 +3,60 @@ import { Table, Th } from 'vayla-design-lib/table/table';
 import { useTranslation } from 'react-i18next';
 import styles from './velho-file-list.scss';
 import { AccordionToggle } from 'vayla-design-lib/accordion-toggle/accordion-toggle';
-import { Button, ButtonVariant } from 'vayla-design-lib/button/button';
 import { formatDateFull } from 'utils/date-utils';
 import InfoboxContent from 'tool-panel/infobox/infobox-content';
 import InfoboxField from 'tool-panel/infobox/infobox-field';
 import { PVDocumentHeader, PVDocumentId } from './velho-model';
-import { useCommonDataAppSelector } from 'store/hooks';
-import { LoaderStatus, useLoaderWithStatus } from 'utils/react-utils';
-import { updateVelhoDocumentsChangeTime } from 'common/change-time-api';
 import { useAppNavigate } from 'common/navigate';
-import { getVelhoDocuments, rejectVelhoDocument } from 'infra-model/infra-model-api';
+import {
+    getVelhoDocuments,
+    rejectVelhoDocument,
+    restoreVelhoDocument,
+} from 'infra-model/infra-model-api';
+import { updateVelhoDocumentsChangeTime } from 'common/change-time-api';
+import { Button, ButtonVariant } from 'vayla-design-lib/button/button';
+import { LoaderStatus, useLoaderWithStatus } from 'utils/react-utils';
+import { TimeStamp } from 'common/common-model';
+
+type ListMode = 'SUGGESTED' | 'REJECTED';
+
+type VelhoFileListContainerProps = {
+    changeTime: TimeStamp;
+    listMode: ListMode;
+};
 
 type VelhoFileListProps = {
     documentHeaders: PVDocumentHeader[];
     isLoading: boolean;
+    listMode: ListMode;
```

Comment:
n ole kauhean tyytyväinen siihen miten täällä tuli lopulta parametrisoitua tuo reject/restore. Käytännössä ainoa ero näiden komponenttien välillä on, että näytetäänkö nappina Hylkää vai Palauta, joten en siksi halunnut eriyttää näitä kokonaan omiksi komponenteikseenkaan. Päädyin lopulta tekemään tämän ListMode-enumeraation, jonka perusteella päätellään mitä tehdään. Not ideal, joten jos joku keksii jotain parempaa, niin olen pelkkiä korvia.

Reply:
Joo vähän hapan, mutta ei hirmu paha. 

Täällä sisällä voisi tietty laittaa toiminnot omiin komponentteihinsa (esim SuggestedFileActions ja RejectedFileActions) jotka pistetään child komponenttina row:lle. Mutta sekään ei poista tätä moodijuttua tältä ylimmältä tasolta koska kunkin rivin napit pitää tehdä erikseen. Jos luonnin passa funkkarina sisään, se menee jo selvästi kamalammaksi härveliksi.

Tietty jos tuohon lisäksi raapisi rivit itsessään omiksi komponenteiksi, voisi tämän käyttäjä luoda listan tyyliin
```
<VelhoFileList ...>{headers.map(h => <VelhoFileListRow ...><VelhoFileListRejectActions />...
```
Mutta sekään ei ole erityisen nättiä. Filelist sanoo kuitenkin headerit ja niiden pitää matchata rivien columneihin -> helpompi pitää ne synkassa kun kaikki on täällä.